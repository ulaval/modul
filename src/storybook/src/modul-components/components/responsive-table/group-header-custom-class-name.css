.test-group-header-custom-class-name.test-group-header-custom-class-name {
    border-left: 4px solid #000;
    background: #ccc;
}

.test-group-header-custom-class-name.test-group-header-custom-class-name th {
    padding: 8px 16px;
}

.test-icon-accordion-custom-class-name.test-icon-accordion-custom-class-name {
    color: blue;
}

.test-cell-custom-class-name.test-cell-custom-class-name {
    background: #000;
    color: #fff;
}
