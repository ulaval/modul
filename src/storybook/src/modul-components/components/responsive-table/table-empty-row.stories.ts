import { actions } from '@storybook/addon-actions';
import { RESPONSIVE_TABLE_NAME, TABLE_EMPTY_ROW_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MEmptyAreaButtonType } from '@ulaval/modul-components/dist/components/empty-area/empty-area';
import { ModulIconName } from '@ulaval/modul-components/dist/utils/modul-icons/modul-icons';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';
import { importAllSvg } from '../svg/svg-importation';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${RESPONSIVE_TABLE_NAME}/${TABLE_EMPTY_ROW_NAME}`,
    parameters: { fileName: __filename }
};

export const Default = () => ({
    data: () => ({
        emptyArea: {
            title: 'Tilte',
            subtitle: 'Subtitle',
            svgName: ModulIconName.Folder,
            buttonText: 'Add content',
            buttonType: MEmptyAreaButtonType.Button
        }
    }),
    methods: actions('emitEmptyButtonClick'),
    beforeCreate() {
        importAllSvg();
    },
    template: `<${TABLE_EMPTY_ROW_NAME}
        :empty-area="emptyArea"
        :table-component-width="'600px'"
        :horizontal-scroll-offset="10"
        :nb-columns="3"
        @empty-button-click="emitEmptyButtonClick"
    />`
});

export const PropWaiting = () => ({
    data: () => ({
        emptyArea: {
            title: 'Text',
            svgName: ModulIconName.Folder,
            buttonText: 'Add content',
            buttonType: MEmptyAreaButtonType.Button
        }
    }),
    methods: actions('emitEmptyButtonClick'),
    beforeCreate() {
        importAllSvg();
    },
    template: `<${TABLE_EMPTY_ROW_NAME}
        :empty-area="emptyArea"
        :table-component-width="'600px'"
        :horizontal-scroll-offset="10"
        :waiting="true"
        @empty-button-click="emitEmptyButtonClick"
    />`
});

export const PropInsideOfTableGroup = () => ({
    data: () => ({
        emptyArea: {
            title: 'Prop inside-of-table-group = true',
            svgName: ModulIconName.Folder
        }
    }),
    beforeCreate() {
        importAllSvg();
    },
    template: `<${TABLE_EMPTY_ROW_NAME}
        :empty-area="emptyArea"
        :table-component-width="'600px'"
        :horizontal-scroll-offset="10"
        :inside-of-table-group="true"
    />`
});
