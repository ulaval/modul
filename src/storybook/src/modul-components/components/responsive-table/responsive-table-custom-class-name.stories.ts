import { select } from '@storybook/addon-knobs';
import { RESPONSIVE_TABLE_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MTableGroupHeaderStyle, MTableRowsStyle } from '@ulaval/modul-components/dist/components/responsive-table/responsive-table-commons';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';
import './group-header-custom-class-name.css';
import { DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE, DEFAULT_TABLE_ROW_1, DEFAULT_TABLE_ROW_2, DEFAULT_TABLE_ROW_3 } from './responsive-table-data';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${RESPONSIVE_TABLE_NAME}/Custom class-name`,
    parameters: { fileName: __filename }
};

export const GroupHeader = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rowGroups: [
            {
                name: 'CustomStyle',
                header: {
                    title: 'Group #1',
                    className: 'test-group-header-custom-class-name'
                },
                accordion: {
                    open: true,
                    iconClassName: 'test-icon-accordion-custom-class-name'
                },
                rows: [
                    DEFAULT_TABLE_ROW_1,
                    DEFAULT_TABLE_ROW_2,
                    DEFAULT_TABLE_ROW_3
                ]
            }
        ]
    }),
    props: {
        groupHeaderStyle: {
            default: select(
                'Prop group-header-style',
                Enums.toValueArray(MTableGroupHeaderStyle),
                MTableGroupHeaderStyle.Any
            )
        },
        rowsStyle: {
            default: select(
                'Prop rows-style',
                Enums.toValueArray(MTableRowsStyle),
                MTableRowsStyle.Borders
            )
        }
    },
    template: `<${RESPONSIVE_TABLE_NAME}
        id="GroupHeader"
        :columns="columns"
        :row-groups="rowGroups"
        :group-header-style="groupHeaderStyle"
        :rows-style="rowsStyle"
    />`
});

export const Cells = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS_DISABLED_SORTABLE,
        rows: [
            {
                cells: {
                    name: {
                        value: 'Vincent Guilmette',
                        className: 'add-custom-class-cell',
                        isHeader: true
                    },
                    id: {
                        value: '111 222 333',
                        className: 'test-cell-custom-class-name'
                    },
                    email: {
                        value: 'vincent.gui@gmail.com'
                    },
                    dateOfBirth: {
                        value: '1976-12-05'
                    },
                    phoneNumber: {
                        value: '819 232-3754'
                    }
                }
            }
        ]
    }),
    props: {
        rowsStyle: {
            default: select(
                'Prop rows-style',
                Enums.toValueArray(MTableRowsStyle),
                MTableRowsStyle.Borders
            )
        }
    },
    template: `<${RESPONSIVE_TABLE_NAME}
        id="Cells"
        :columns="columns"
        :rows="rows"
        :rows-style="rowsStyle"
    />`
});
