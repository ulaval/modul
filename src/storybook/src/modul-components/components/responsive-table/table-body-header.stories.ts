import { boolean, object, select, text } from '@storybook/addon-knobs';
import { RESPONSIVE_TABLE_NAME, TABLE_GROUP_HEADER_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MTableColumn, MTableEmptyArea, MTableGroup, MTableGroupHeaderStyle } from '@ulaval/modul-components/dist/components/responsive-table/responsive-table-commons';
import { Enums } from '@ulaval/modul-components/dist/utils/enums/enums';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';
import { DEFAULT_TABLE_COLUMNS, DEFAULT_TABLE_GROUP_1 } from './responsive-table-data';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${RESPONSIVE_TABLE_NAME}/${TABLE_GROUP_HEADER_NAME}`,
    parameters: { fileName: __filename }
};

export const defaultStory = () => ({
    data: () => ({
        columns: DEFAULT_TABLE_COLUMNS,
        rowsGroup: DEFAULT_TABLE_GROUP_1
    }),
    template: `<table>
        <${TABLE_GROUP_HEADER_NAME}
            :columns="columns"
            :rowsGroup="rowsGroup"
        />
   </table>`
});

defaultStory.story = {
    name: 'default'
};

export const Sandbox = () => ({
    props: {
        firstColumnFixed: {
            default: boolean('Prop first-column-fixed', false)
        },
        tableComponentWidth: {
            default: text('Prop table-component-width', '100%')
        },
        groupHeaderStyle: {
            default: select(
                'Prop group-header-style',
                Enums.toValueArray(MTableGroupHeaderStyle),
                MTableGroupHeaderStyle.Light
            )
        },
        columns: {
            default: object<MTableColumn[]>('Prop columns', DEFAULT_TABLE_COLUMNS)
        },
        rowsGroup: {
            default: object<MTableGroup>('Prop rows-group', DEFAULT_TABLE_GROUP_1)
        },
        hasDataRowsGroup: {
            default: boolean('has data rows group', true)
        },
        slotBodyHeaderTitle: {
            default: boolean('Slot body-header-title', false)
        },
        slotBodyHeader: {
            default: boolean('Slot body-header', false)
        },
        defaultEmptyArea: {
            default: object<MTableEmptyArea>('Prop default-empty-area', {
                headerText: 'Default empty area: headerText',
                text: 'Default empty area: text',
                iconName: 'm-svg__clock'
            })
        }
    },
    computed: {
        rowsGroupIntern(): MTableGroup[] {
            const _this: any = this;
            return _this.hasDataRowsGroup ? _this.rowsGroup : [];
        }
    },
    template: `<table style="width: 100%">
        <${TABLE_GROUP_HEADER_NAME}
            :columns="columns"
            :rows-group="rowsGroupIntern"
            :first-column-fixed="firstColumnFixed"
            :group-header-style="groupHeaderStyle"
            :default-empty-area="defaultEmptyArea"
            :table-component-width="tableComponentWidth"
        >
            <template
                v-if="slotBodyHeaderTitle && rowsGroup && rowsGroup.header && rowsGroup.header.title"
                slot="body-header-title"
                slot-scope="{ rowsGroup }">
                {{ rowsGroup.header.title}} (Slot body-header-title)
            </template>
            <template
                v-if="slotBodyHeader"
                v-for="(column, columnIndex) in columns"
                :slot="'body-header.' + column.id"
                slot-scope="{ rowsGroup }"
            >
                {{ rowsGroup.header.cells[column.id].value}} (Slot body-header.{{column.id}})
            </template>
        </${TABLE_GROUP_HEADER_NAME}>
    </table>`
});
