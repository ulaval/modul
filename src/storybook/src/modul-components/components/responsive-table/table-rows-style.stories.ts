import { RESPONSIVE_TABLE_NAME } from '@ulaval/modul-components/dist/components/component-names';
import { MTableRowsStyle } from '@ulaval/modul-components/dist/components/responsive-table/responsive-table-commons';
import Vue, { VueConstructor } from 'vue';
import { modulComponentsHierarchyRootSeparator } from '../../../utils';
import { DEFAULT_TABLE_HEAD_ROWS, DEFAULT_TABLE_ROWS } from './responsive-table-data';

export default {
    title: `${modulComponentsHierarchyRootSeparator}${RESPONSIVE_TABLE_NAME}/Prop rows-style`,
    parameters: { fileName: __filename }
};

const COMPLEX_HEAD_ROWS_TEXT: string = '(complex head-rows)';

const getTemplate: (
    rowsStyle: MTableRowsStyle
) => VueConstructor = (
    rowsStyle: MTableRowsStyle
): VueConstructor => {
    return Vue.extend({
        data: () => ({
            rowsStyle,
            headRows: DEFAULT_TABLE_HEAD_ROWS,
            rows: DEFAULT_TABLE_ROWS
        }),
        template: `<${RESPONSIVE_TABLE_NAME}
            :id="rowsStyle"
            :head-rows="headRows"
            :rows="rows"
            :rows-style="rowsStyle"
        />`
    });
};

export const AlternateBackground = () => {
    return getTemplate(MTableRowsStyle.AlternateBackground);
};

export const Borders = () => {
    return getTemplate(MTableRowsStyle.Borders);
};

export const CellBorders = () => {
    return getTemplate(MTableRowsStyle.CellBorders);
};
