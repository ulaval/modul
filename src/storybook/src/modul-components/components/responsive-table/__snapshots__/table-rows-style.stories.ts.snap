// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots modul-components|m-responsive-table/Prop rows-style Alternate Background 1`] = `
<div
  class="m-auto-horizontal-scroll m-responsive-table m--is-gradient-style-white"
  style="position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 100%;"
    >
      <table
        cellspacing="0"
        class="m-responsive-table__table"
        id="idTable"
      >
        <thead
          class="m-table-head m--is-head-style-light"
        >
          <tr>
            <th
              class="m--is-sortable m--is-sorted m--is-text-align-left"
              id="name-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Name
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple m--is-sort-asc"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
            <th
              class="m--is-sortable m--is-text-align-center"
              id="id-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Id
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
            <th
              class="m--is-sortable m--is-text-align-left"
              id="email-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Email
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
            <th
              class="m--is-sortable m--is-text-align-left"
              id="dateOfBirth-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Date of birth
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
            <th
              class="m--is-sortable m--is-text-align-left"
              id="phoneNumber-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Phone number
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
          </tr>
        </thead>
         
        <!---->
         
        <tbody
          class="m-table-group m--no-header m--has-row-highlighted-on-hover m--is-rows-style-alternate-background"
        >
          <!---->
           
          <tr>
            <th
              class="m-table-group__cell m--is-text-align-left add-custom-class-cell"
              scope="row"
            >
              
                                Name 1
                            
            </th>
            <td
              class="m-table-group__cell m--is-text-align-center"
            >
              
                                111 222 333
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                fake1@email.com
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                1971-12-05
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                819 555-0111
                            
            </td>
          </tr>
          <tr>
            <th
              class="m-table-group__cell m--is-text-align-left"
              scope="row"
            >
              
                                Name 2
                            
            </th>
            <td
              class="m-table-group__cell m--is-text-align-center"
            >
              
                                111 322 344
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                fake2@email.com
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                1980-06-11
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                418 555-0163
                            
            </td>
          </tr>
          <tr>
            <th
              class="m-table-group__cell m--is-text-align-left"
              scope="row"
            >
              
                                Name 3
                            
            </th>
            <td
              class="m-table-group__cell m--is-text-align-center"
            >
              
                                112 528 967
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                fake3@email.com
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                1970-04-08
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                438 555-0113
                            
            </td>
          </tr>
           
          <!---->
        </tbody>
         
        <!---->
      </table>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots modul-components|m-responsive-table/Prop rows-style Borders 1`] = `
<div
  class="m-auto-horizontal-scroll m-responsive-table m--is-gradient-style-white"
  style="position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 100%;"
    >
      <table
        cellspacing="0"
        class="m-responsive-table__table"
        id="idTable"
      >
        <thead
          class="m-table-head m--is-head-style-light"
        >
          <tr>
            <th
              class="m--is-sortable m--is-sorted m--is-text-align-left"
              id="name-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Name
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple m--is-sort-asc"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
            <th
              class="m--is-sortable m--is-text-align-center"
              id="id-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Id
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
            <th
              class="m--is-sortable m--is-text-align-left"
              id="email-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Email
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
            <th
              class="m--is-sortable m--is-text-align-left"
              id="dateOfBirth-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Date of birth
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
            <th
              class="m--is-sortable m--is-text-align-left"
              id="phoneNumber-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Phone number
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
          </tr>
        </thead>
         
        <!---->
         
        <tbody
          class="m-table-group m--no-header m--has-row-highlighted-on-hover m--is-rows-style-borders"
        >
          <!---->
           
          <tr>
            <th
              class="m-table-group__cell m--is-text-align-left add-custom-class-cell"
              scope="row"
            >
              
                                Name 1
                            
            </th>
            <td
              class="m-table-group__cell m--is-text-align-center"
            >
              
                                111 222 333
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                fake1@email.com
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                1971-12-05
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                819 555-0111
                            
            </td>
          </tr>
          <tr>
            <th
              class="m-table-group__cell m--is-text-align-left"
              scope="row"
            >
              
                                Name 2
                            
            </th>
            <td
              class="m-table-group__cell m--is-text-align-center"
            >
              
                                111 322 344
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                fake2@email.com
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                1980-06-11
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                418 555-0163
                            
            </td>
          </tr>
          <tr>
            <th
              class="m-table-group__cell m--is-text-align-left"
              scope="row"
            >
              
                                Name 3
                            
            </th>
            <td
              class="m-table-group__cell m--is-text-align-center"
            >
              
                                112 528 967
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                fake3@email.com
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                1970-04-08
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                438 555-0113
                            
            </td>
          </tr>
           
          <!---->
        </tbody>
         
        <!---->
      </table>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots modul-components|m-responsive-table/Prop rows-style Cell Borders 1`] = `
<div
  class="m-auto-horizontal-scroll m-responsive-table m--is-gradient-style-white"
  style="position: relative;"
>
  <!---->
   
  <!---->
   
  <div
    class="m-auto-horizontal-scroll__body"
  >
    <div
      class="m-auto-horizontal-scroll__body-content"
      style="min-width: 100%;"
    >
      <table
        cellspacing="0"
        class="m-responsive-table__table"
        id="idTable"
      >
        <thead
          class="m-table-head m--is-head-style-light"
        >
          <tr>
            <th
              class="m--is-sortable m--is-sorted m--is-text-align-left"
              id="name-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Name
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple m--is-sort-asc"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
            <th
              class="m--is-sortable m--is-text-align-center"
              id="id-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Id
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
            <th
              class="m--is-sortable m--is-text-align-left"
              id="email-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Email
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
            <th
              class="m--is-sortable m--is-text-align-left"
              id="dateOfBirth-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Date of birth
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
            <th
              class="m--is-sortable m--is-text-align-left"
              id="phoneNumber-idTable"
              scope="col"
            >
              <div
                class="m-table-head__cell"
              >
                <div
                  class="m-table-head__cell-content"
                >
                  <div>
                    Phone number
                  </div>
                   
                  <button
                    class="m-icon-button m-table-head__sort-icon m--is-skin-light m--has-ripple"
                    style="width: 14px; height: 14px;"
                    type="button"
                  >
                    <svg
                      aria-hidden="true"
                      class="m-icon m-icon-button__icon"
                      height="12px"
                      width="12px"
                    >
                      <!---->
                       
                      <use
                        aria-hidden="true"
                        class=""
                      />
                    </svg>
                     
                    <!---->
                  </button>
                </div>
              </div>
            </th>
          </tr>
        </thead>
         
        <!---->
         
        <tbody
          class="m-table-group m--no-header m--has-row-highlighted-on-hover m--is-rows-style-cell-borders"
        >
          <!---->
           
          <tr>
            <th
              class="m-table-group__cell m--is-text-align-left add-custom-class-cell"
              scope="row"
            >
              
                                Name 1
                            
            </th>
            <td
              class="m-table-group__cell m--is-text-align-center"
            >
              
                                111 222 333
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                fake1@email.com
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                1971-12-05
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                819 555-0111
                            
            </td>
          </tr>
          <tr>
            <th
              class="m-table-group__cell m--is-text-align-left"
              scope="row"
            >
              
                                Name 2
                            
            </th>
            <td
              class="m-table-group__cell m--is-text-align-center"
            >
              
                                111 322 344
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                fake2@email.com
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                1980-06-11
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                418 555-0163
                            
            </td>
          </tr>
          <tr>
            <th
              class="m-table-group__cell m--is-text-align-left"
              scope="row"
            >
              
                                Name 3
                            
            </th>
            <td
              class="m-table-group__cell m--is-text-align-center"
            >
              
                                112 528 967
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                fake3@email.com
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                1970-04-08
                            
            </td>
            <td
              class="m-table-group__cell m--is-text-align-left"
            >
              
                                438 555-0113
                            
            </td>
          </tr>
           
          <!---->
        </tbody>
         
        <!---->
      </table>
    </div>
  </div>
  <div
    class="resize-sensor"
    dir="ltr"
    style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
  >
    <div
      class="resize-sensor-expand"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s;"
      />
    </div>
    <div
      class="resize-sensor-shrink"
      style="pointer-events: none; position: absolute; left: 0px; top: 0px; right: 0px; bottom: 0px; overflow: hidden; z-index: -1; visibility: hidden; max-width: 100%;"
    >
      <div
        style="position: absolute; left: 0px; top: 0px; transition: 0s; width: 200%; height: 200%;"
      />
    </div>
  </div>
</div>
`;
